### PR TITLE
CRIMAPP-1230 Structs support draft applications

### PR DESCRIPTION
**Pull request by timpeat**
## Description of change

v 1.2.0
Structs support draft applications. 

## Link to relevant ticket
[CRIMAPP-1230](https://dsdmoj.atlassian.net/browse/CRIMAPP-1230)

## Additional notes

This change is to allow them to be used an Apply for draft applications (see: https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/pull/1085 )

[CRIMAPP-1230]: https://dsdmoj.atlassian.net/browse/CRIMAPP-1230?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: lib/laa_crime_schemas/structs/saving.rb [hunk 1]
```diff
@@ -4,14 +4,14 @@ module LaaCrimeSchemas
   module Structs
     class Saving < Base
       attribute :saving_type, Types::SavingType
-      attribute :provider_name, Types::String
+      attribute? :provider_name, Types::String.optional
       attribute :account_balance, Types::PenceSterling
```
Comment (hiboabd) [inline lib/laa_crime_schemas/structs/saving.rb:8]: Is this meant to be optional too?

Comment (hiboabd) [inline lib/laa_crime_schemas/structs/saving.rb:8]: Is this meant to be optional too?

File: lib/laa_crime_schemas/structs/investment.rb [hunk 2]
```diff
@@ -4,8 +4,8 @@ module LaaCrimeSchemas
   module Structs
     class Investment < Base
       attribute :investment_type, Types::InvestmentType
```
Comment (hiboabd) [inline lib/laa_crime_schemas/structs/investment.rb:6]: Is this meant to be optional?

File: lib/laa_crime_schemas/structs/saving.rb (hunk 1 above)
Comment (timpeat) [reply lib/laa_crime_schemas/structs/saving.rb:8]: Yes, the json-schema makes sure this is required on submission, but can be nil in draft.

File: lib/laa_crime_schemas/structs/investment.rb (hunk 2 above)
Comment (timpeat) [reply lib/laa_crime_schemas/structs/investment.rb:6]: This can xxx_type can never not exist as there is a database constraint on Apply

**Review by hiboabd — APPROVED** [review]
🚀

**Pull request merged**